test/test-execute/exec-unset-environment.service

```diff
@@ -0,0 +1,8 @@
+[Unit]
+Description=Test for UnsetEnvironment
+
+[Service]
+ExecStart=/bin/sh -x -c 'test "$$FOO" = "bar" && test "$$QUUX" = "" && test "$$VAR3" = "value3" && test "$$VAR4" = "" && test "$$VAR5" = "value5" && test "$$X%b" = ""'
```

Comment:
I'm not sure that `test "$VAR" = ""` can be used to check whether `$VAR` is unset or not. Would it make more sense to use `test "${VAR-X} = "X"` instead?

Reply:
Similarly, to check that `VAR` is set and has an empty value, something like `test ${VAR?} = ""` could be used.

Reply:
i thought about using that, but i think the ${VAR-X} syntax is a bashism and so far we run our tests with plain sh... ultimately i gave up as this is messy in bourne shell and just do what most people do...

Reply:
I think it is safe to use that, because `-` and `?` are mentioned in http://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html. `dash` and `busybox` support such a syntax too.

Reply:
ah, very good, will make such a change